Library/Homebrew/test/dependency_expansion_spec.rb

```diff
@@ -101,18 +101,6 @@ def build_dep(name, tags = [], deps = [])
     expect(deps).to eq([bar, baz])
   end
 
-  it "keeps dependency but prunes recursive dependencies with ::keep_but_prune_recursive_deps" do
```

Comment:
Could you write/modify a test for the reverse instead i.e. verify that these aren't pruned?

Reply:
This test was removed not because of fix itself but because the method it was using was removed.

I tried to (by making a test which should fail with old build.rb but pass with new one), but it didn't work.
I guess it should be in some other place, since the change to be tested is actually `Build#expand_deps` and not something in `dependency_expansion`

Reply:
> I guess it should be in some other place, since the change to be tested is actually Build#expand_deps and not something in dependency_expansion

👍 That's be great.

Reply:
But I'm not sure on how to test that -- I guess a new spec needs to be created for "Build"?

Reply:
@stek29 Actually you can leave this as-is; I will create another PR that plays around a bit with this first. Nice work again!

Reply:
@MikeMcQuaid Sure.